app/components/FormFields/TextBlock.jsx

```diff
@@ -15,13 +15,35 @@ export const TextBlock = ({ content }) => (
       ul li {
         list-style-type: disc;
       }
+      iframe {
+        max-width: 100%;
+      }
       min-width: 0;
     `}
     warpperElement={{ 'data-color-mode': 'light' }}
-    rehypeRewrite={(node) => {
+    rehypeRewrite={(node, index, parent) => {
       if (node.tagName === 'input') {
         node.properties.disabled = false;
       }
+      if (
+        parent?.tagName === 'a' &&
+        parent?.properties.href.slice(0, 24) == 'https://www.youtube.com/'
+      ) {
+        const youtubeLink = `https://www.youtube.com/embed/${node.value.slice(
+          32,
+        )}`;
+        node.value = (
+          <iframe
```

Comment:
the same code is duplicated in the components, take it out into a separate component and reuse

Reply:
done: https://github.com/peeranha/peeranha-web/pull/1268/commits/5c12067e1cc582383f72be0d7ec97006c582ac19